vowpalwabbit/slates_label.cc

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "slates_label.h"
+
+#include "cache.h"
+#include "parser.h"
+#include "vw_string_view.h"
+
+#include <numeric>
+
+namespace slates
+{
+void default_label(void* v);
+
+#define READ_CACHED_VALUE(DEST, TYPE)                            \
```

Comment:
Is there a reason these are macros and not inlined functions? Because of the early return?

Reply:
Yep the early return was the main reason. Then for the other one, since it is so small, I thought keeping it consistent with this macro made sense

Reply:
Yep the early return was the main reason. Then for the other one, since it is so small, I thought keeping it consistent with this macro made sense